book.rb

```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.classroom = self
+  end
```

Comment:
- [ ] Because of the above refactoring in the rental class. The add rental method should change a little bit. It will need two parameters. That is the person renting the book, and the data.

    Another issue is that the `Rentals class` takes care of pushing the `rental object` to `book.rentals`. Therefore, this method needs no repeat the same step of pushing the rental. In this method, we only need to create the Rental.

```ruby
def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
end
```

## Some correction
This line `rental.classroom = self` could have not been correct since rental has not classroom and does not belong to any.